### PR TITLE
Update readme.md

### PR DESCRIPTION
**Pull request by rohitnz**
Update hyperlinks in readme.md so they are in-line and correct typos

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -667,18 +660,17 @@ Array.method('reduce', function (parameters){
 * Accessing a missing element will give you `undefined`
 * If you have an algorithm that relies on the array not being empty and not having `undefined` values, you can write a function that will prep your array to have a certain number of defined values, essentially initializing it with certain values in place
 	* An `Array.dim` function is oulined on page 63 which will allow `var myArray = Array.dim(10,0)` to make an array with 10 zeroes starting from the first position in the array(0)
-* Javascript only has **one dimensional arrays** but **_can_ have arrays of arrays**
-* Two dimensional arrays (matricess) will have to be set up by the programmer
+* Javascript only has **one dimensional arrays** but **_can_ have array of arrays**
```
Comment (iteles) [inline readme.md:663]: I don't think this one was a typo, but I also don't think it changes the meaning here so no problem! 👍

**Review by iteles — APPROVED** [review]
WOW 😱 

I'm so impressed by how thoroughly you've read these notes @rohitnz ! 

Thank you so much for all of these fantastic contributions 🎉 

I've also invited you to the @dwyl organisation where we have lots of great open source `learn-xxxxxxx` tutorials for all your coding needs 😉 Best place to start if you're interested is putting up a profile picture (any one!)  and https://github.com/dwyl/start-here plus the issues in that repo!

File: readme.md (hunk 1 above)
Comment (nelsonic) [reply readme.md:663]: I _guess_ this could be clarified by saying "_nested arrays_" ... but think this is OK. 😉

**Review by nelsonic — APPROVED** [review]
@rohitnz _great work_ tidying up the README! 🎉

**Comment by rohitnz** [comment]
@iteles Thank you. Yes, my mistake, 'arrays of arrays' was alright. I wanted to quickly pickup some good practices from this book over the weekend. Google search led me to this repo. You did an excellent job condensing an already terse material. Thank you for the invitation but I am a novice.
@nelsonic Thank you.

**Pull request merged**